pkg/extensions/sync/service.go

```diff
@@ -60,16 +61,42 @@ func New(
 	var err error
 
 	var credentialsFile syncconf.CredentialsFile
-	if credentialsFilepath != "" {
-		credentialsFile, err = getFileCredentials(credentialsFilepath)
-		if err != nil {
-			log.Error().Str("errortype", common.TypeOf(err)).Str("path", credentialsFilepath).
-				Err(err).Msg("couldn't get registry credentials from configured path")
+
+	if service.config.CredentialHelper == "" {
```

Comment:
Instead of a nested double-if, negate this check as if ...CredentialHelper != "" {} else {//check file}